public/javascripts/items.js

```diff
@@ -10,25 +10,83 @@ const fetchParams = {
 const params = completed =>
   Object.assign( {}, fetchParams, { body: JSON.stringify({ completed }) })
 
+const params2 = title =>
+  Object.assign( {}, fetchParams, { body: JSON.stringify({ title }) })
+
+const params3 = description =>
+  Object.assign( {}, fetchParams, { body: JSON.stringify({ description }) })
+
 $(document).ready( () => {
-  $( '.title' ).click( event => {
+  $('.updateTitle').keypress( event => {
+    element = $( event.target )
+    const id = element.data( 'id' )
+
+    if( event.charCode === 13 ) {
+      let updatedTitle = element[0].value
+      fetch( `/items/${id}/edit`, params2( updatedTitle ))
+      .then( result => result.json() )
+      .then( json => {
+        if( json.success ) {
+          let span = element.prev()
+          $( span[0] ).html(updatedTitle)
+          element.addClass( 'hidden')
+          $( span ).removeClass( 'hidden' )
+        }
+      })
+    }
+  })
+
+  $( '.title > span' ).click( event => {
     const element = $( event.target )
+    $( element[0] ).addClass( 'hidden' )
+    let input = element.next()
+    $( input[0]).removeClass( 'hidden' )
+  })
+
+  $('.updateDescription').keypress( event => {
+    element = $( event.target )
     const id = element.data( 'id' )
-    const completed = ! element.data( 'completed' )
 
-    fetch( `/items/${id}/completed`, params( completed ))
+    if( event.charCode === 13 ) {
+      let updatedDescription = element[0].value
+      fetch( `/items/${id}/edit`, params3( updatedDescription ))
       .then( result => result.json() )
       .then( json => {
         if( json.success ) {
-          element.data( 'completed', completed )
-
-          const parent = $( `.item[data-id=${id}]` )
-          if( completed ) {
-            parent.addClass( 'completed' )
-          } else {
-            parent.removeClass( 'completed' )
-          }
+          let span = element.prev()
+          $( span[0] ).html(updatedDescription)
+          element.addClass( 'hidden')
+          $( span ).removeClass( 'hidden' )
         }
       })
+    }
+  })
+
+  $( '.description > span' ).click( event => {
+    const element = $( event.target )
```

Comment:
@jrob8577 
Ethan: "I can see creating a function like `getElementAndId`, or perhaps `getElement` and `getId` to avoid returning a confusing array. Is this what you were talking about?